tools/Projects/src/Views/LauncherView.xaml.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Text;
+using DevHome.Projects.ViewModels;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using static Windows.Win32.PInvoke;
+
+namespace DevHome.Projects.Views;
+public sealed partial class LauncherView : UserControl
+{
+    public LauncherView()
+    {
+        this.InitializeComponent();
+    }
+
+    private void LauncherButton_Click(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
+    {
+        var project = (sender as Button).DataContext as LauncherViewModel;
+        project?.Launch();
+    }
+
+    private LauncherViewModel ViewModel => (LauncherViewModel)this.DataContext;
+
+    private string ResolvePath(string inputPath, string launcherDirectory = null)
+    {
+        var cwd = ViewModel.ProjectViewModel.TryGetTarget(out var project) ? project.ExpandedFilePath : string.Empty;
+        Span<char> arr = stackalloc char[1024];

```

Comment:
1kb is a bit big for a stack allocation and PathFindOnPath explicitly says that the limit in length is MAX_PATH. Why not use MAX_PATH (260) + 1 here?